### PR TITLE
small fix

### PR DESCRIPTION
**Pull request by fan-zhang-sv**

### CODE REVIEW

File: packages/wallet-sdk/src/CoinbaseWalletProvider.ts [hunk 1]
```diff
@@ -44,6 +44,7 @@ export class CoinbaseWalletProvider extends EventEmitter implements ProviderInte
     // eth_requestAccounts
     handshake: async (_: RequestArguments): Promise<AddressString[]> => {
       if (this.connected) {
+        this.emit('connect', { chainId: prepend0x(this.chain.id.toString(16)) });
```
Comment (fan-zhang-sv) [inline packages/wallet-sdk/src/CoinbaseWalletProvider.ts:47]: turns out we still need this, when dapp reloads and they dont rememebr logged in state, we'd need to alert them

Comment (bangtoven) [inline packages/wallet-sdk/src/CoinbaseWalletProvider.ts:47]: i thought we didn't emit `connect` for already connected case?

File: packages/wallet-sdk/src/sign/SignHandler.ts [hunk 2]
```diff
@@ -42,6 +41,7 @@ export class SignHandler {
       url: keysUrl ?? CB_KEYS_URL,
       onConfigUpdateMessage: this.handleConfigUpdateMessage.bind(this),
     });
+    this.signer = this.loadSigner();
```
Comment (fan-zhang-sv) [inline packages/wallet-sdk/src/sign/SignHandler.ts:44]: init signer needs the metadata, etc assigned

Comment (bangtoven) [inline packages/wallet-sdk/src/sign/SignHandler.ts:44]: is there any issue with sequence of these lines?

**Review by bangtoven — COMMENTED** [review]
@fan-zhang-sv

**Review by bangtoven — APPROVED** [review]

File: packages/wallet-sdk/src/CoinbaseWalletProvider.ts (hunk 1 above)
Comment (fan-zhang-sv) [reply packages/wallet-sdk/src/CoinbaseWalletProvider.ts:47]: the playground testdapp is one example

it loses connected state on reload, and it needs this connect message to enable the rest of functionalities

File: packages/wallet-sdk/src/sign/SignHandler.ts (hunk 2 above)
Comment (fan-zhang-sv) [reply packages/wallet-sdk/src/sign/SignHandler.ts:44]: yea, loadSigner reads this.metadata, etc, which was undefined

**Pull request merged**